src/OrchardCore.Modules/OrchardCore.Settings/Drivers/DefaultSiteSettingsDisplayDriver.cs

```diff
@@ -14,33 +14,40 @@ public class DefaultSiteSettingsDisplayDriver : DisplayDriver<ISite>
         public const string GroupId = "general";
 
         protected readonly IStringLocalizer S;
+
         private readonly IShellHost _shellHost;
         private readonly ShellSettings _shellSettings;
 
-        public DefaultSiteSettingsDisplayDriver(IStringLocalizer<DefaultSiteSettingsDisplayDriver> stringLocalizer,
+        public DefaultSiteSettingsDisplayDriver(
             IShellHost shellHost,
-            ShellSettings shellSettings)
+            ShellSettings shellSettings,
+            IStringLocalizer<DefaultSiteSettingsDisplayDriver> stringLocalizer)
         {
-            S = stringLocalizer;
             _shellHost = shellHost;
             _shellSettings = shellSettings;
+            S = stringLocalizer;
         }
 
-        public override IDisplayResult Edit(ISite site)
+        public override Task<IDisplayResult> EditAsync(ISite site, BuildEditorContext context)
         {
-            return Initialize<SiteSettingsViewModel>("Settings_Edit", model =>
-            {
-                model.SiteName = site.SiteName;
-                model.PageTitleFormat = site.PageTitleFormat;
-                model.BaseUrl = site.BaseUrl;
-                model.TimeZone = site.TimeZoneId;
-                model.PageSize = site.PageSize;
-                model.UseCdn = site.UseCdn;
-                model.CdnBaseUrl = site.CdnBaseUrl;
-                model.ResourceDebugMode = site.ResourceDebugMode;
-                model.AppendVersion = site.AppendVersion;
-                model.CacheMode = site.CacheMode;
-            }).Location("Content:1").OnGroup(GroupId);
+            context.Shape.Metadata.Wrappers.Add("GeneralSettingsWrapper");
+
+            var result = Combine(
+
+                Initialize<SiteSettingsViewModel>("Settings_Edit", model => PopulateProperties(site, model))
+                    .Location("Content:1#Site;10")
+                    .OnGroup(GroupId),
+
+                Initialize<SiteSettingsViewModel>("SettingsResources_Edit", model => PopulateProperties(site, model))
```

Comment:
```suggestion
                Initialize<SiteSettingsViewModel>("Settings_Resources_Edit", model => PopulateProperties(site, model))
```

Reply:
not sure what would the file name be if we changed it as suggested. `SettingsResources_Edit` > `Settings_Resources.Edit.cshtml`.

Reply:
As I remember `_` becomes a dot `.` and a double underscores `__` becomes a dash `-`.

So `Settings__Resources_Edit` with double underscores would have been resolved to `Settings-Resources.Edit`.

But I think that here `_Edit` is considered as the display type, if so the convention is to place the display type just after the shape base type, this because the display type is discovered as the first element separated with a `_`, this allows to add other elements separated with a `_`,  then when converted to a file name it is moved to the end.

So `Settings_Edit__Resources` would have been the right template name and still resolved to `Settings-Resources.Edit`.

Reply:
Thanks you! I'll try it

Reply:
@jtkech what about the wrapper name? Would we change `GeneralSettingsWrapper`? How can we generate `GeneralSettings.Wrapper.cshtml`?

Reply:
Only from memory so should be tried.

`GeneralSettings_Wrapper` => `GeneralSettings.Wrapper`

`GeneralSettings_Wrapper__OtherTerm` => `GeneralSettings-OtherTerm.Wrapper`

Hmm, maybe in TheBlogTheme there is a wrong file name `Widget.Wrapper-Zone-HeadMeta.liquid`, maybe it should be `Widget-Zone-HeadMeta.Wrapper.liquid`.

Reply:
not sure what that shape is or does. maybe a separate PR.

Reply:
Yes, was just for info when I saw it